.github/workflows/gradle.yml

```diff
@@ -394,7 +430,9 @@ jobs:
           ec2-instance-id: ${{ needs.start-platform-build-runner.outputs.ec2-instance-id }}
 
   octavia-cli-build:
+    needs: changes
```

Comment:
very nit: reading this here I think it would be clearer that `changes` is another build step and not GitHub action magic with a more clear name for it's action, ie: `check-for-directory-changes`

Reply:
good point. renaming.

Reply:
actually changed my mind - this would make all the references to this action very long so the various if conditionals would get less readable. I'm hoping the description is sufficient. we can come back to this if you feel strongly.